sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -182,11 +183,16 @@ case class CaseWhen(branches: Seq[(Expression, Expression)], elseValue: Option[E
 
     generatedCode += "}\n" * cases.size
 
-    s"""
+    // Methods that are too long cannot be compiled, so fall back to interpreting
+    if (generatedCode.length > 1000) {
```

Comment:
Can we do the switch based on number of branches? for example, 10 or 20 switches. 

Because whole stage codegen can't work with CodegenFallback, If we do so, we can know when we will do the fallback, we can still support CaseWhen in whole stage codegen if only have a few branches.